config/routes.rb

```diff
@@ -24,6 +24,7 @@
   concern :site do
     resources :posts, only: %i(show), constraints: {id: /\d+/}, path: '/'
```

Comment:
By the way, we should compatible with old URL for Posts#show.
Could you add this line?

```
get '/:id/:page' => redirect('/%{id}?page=%{page}'), constraints: {id: /\d+/, page: /\d+/} # To compatible with old URL
```